newIDE/app/src/Utils/GridHelpers.js

```diff
@@ -1,12 +1,61 @@
 // @flow
 
 export const roundPosition = (
-  pos: number,
-  gridSize: number,
-  gridOffset: number
+  pos: [number, number],
+  gridWidth: number,
+  gridHeight: number,
+  gridOffsetX: number,
+  gridOffsetY: number,
+  gridType: number
 ) => {
-  if (gridSize <= 0) {
-    return Math.round(pos);
+  if (gridType === 'isometric') {
+    if (gridWidth <= 0 || gridHeight <= 0) {
+      pos[0] = Math.round(pos[0]);
+      pos[1] = Math.round([1]);
+      return;
+    }
+    // Why do we need this?
+    // Take a 2x2 squares, put a diamond inside each square, there is a 5th diamond in the center
```

Comment:
This comment is not really explaining the calculation, could you add something explaining what cellX/Y represents?